tests/testthat/test-render.R

```diff
@@ -203,3 +203,76 @@ test_that("render_book() works - chapters", {
 
 })
 
+test_that("render_website() works - listing", {
+  withr::local_envvar("BABELQUARTO_TESTS_URL" = "true")
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+  quarto_multilingual_website(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en"
+  )
+
+  fs::dir_create(file.path(parent_dir, project_dir, "subdir"))
+  listing_lines <- c(
+    "---",
+    'title: "Listing"',
+    "listing:",
+    '  - contents: ["subdir/*.qmd"]',
+    "    type: grid",
+    "---"
+  )
+  brio::write_lines(
+    listing_lines,
+    file.path(parent_dir, project_dir, "listing.qmd")
+  )
+  fs::file_copy(
+    file.path(parent_dir, project_dir, "about.qmd"),
+    file.path(parent_dir, project_dir, "subdir", "about.qmd")
+  )
+  fs::file_copy(
+    file.path(parent_dir, project_dir, "about.fr.qmd"),
+    file.path(parent_dir, project_dir, "subdir", "about.fr.qmd")
+  )
+
+  withr::with_dir(parent_dir, render_website(project_dir))
+  
+  index <- xml2::read_html(file.path(parent_dir, project_dir, "_site", "listing.html"))
+  listing_grid <- xml2::xml_find_first(index, "//div[contains(@class, 'quarto-listing-cols-3')]")
+  grid_items <- xml2::xml_find_all(listing_grid, ".//div[contains(@class, 'quarto-grid-item')]")
+
+  expect_length(grid_items, 1)
+})
+
+test_that("render_website() works - clean render for each language", {
+  withr::local_envvar("BABELQUARTO_TESTS_URL" = "true")
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+  quarto_multilingual_website(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en"
+  )
+
+  fs::dir_create(file.path(parent_dir, project_dir, "subdir"))
+  fs::file_copy(
+    file.path(parent_dir, project_dir, "about.qmd"),
+    file.path(parent_dir, project_dir, "subdir", "about.qmd")
+  )
+  fs::file_copy(
+    file.path(parent_dir, project_dir, "about.fr.qmd"),
+    file.path(parent_dir, project_dir, "subdir", "about.fr.qmd")
+  )
+
+  withr::with_dir(parent_dir, render_website(project_dir))
+  expect_true(fs::dir_exists(file.path(parent_dir, project_dir, "_site", "subdir")))
```

Comment:
not for this PR but we might think about adding helpers in a helper file https://testthat.r-lib.org/articles/special-files.html#helper-files such as `expect_dir_exists()` and `expect_file_exists()` if the pattern is used a lot.

```r
expect_dir_exists <- function(path) {
  expect_true(fs::dir_exists(path))
}

```